app/models/temp_file.rb

```diff
@@ -3,4 +3,12 @@ class TempFile < ActiveRecord::Base
 
   has_attached_file :file
   do_not_validate_attachment_file_type :file
+
+  def destroy_obsolete_files
+    file = TempFile.find_by_id(self.id)
+    file.destroy! if file
```

Comment:
If the user complete the import of samples the temp file is deleted. Here I check if the TempFile is still present to prevent the call destroy on nil class because this job is triggered on every upload.